consensus/hotstuff/votecollector/combined_vote_processor_v2.go

```diff
@@ -259,20 +259,43 @@ func (p *CombinedVoteProcessorV2) buildQC() (*flow.QuorumCertificate, error) {
 		return nil, fmt.Errorf("could not reconstruct random beacon group signature: %w", err)
 	}
 
-	blockSigData := &hotstuff.BlockSignatureData{
+	blockSigData := buildBlockSignatureDataForV2(stakingSigners, aggregatedStakingSig, reconstructedBeaconSig)
+
+	qc, err := buildQCWithPackerAndSigData(p.packer, p.block, blockSigData)
+	if err != nil {
+		return nil, err
+	}
+
+	return qc, nil
+}
+
+// buildBlockSignatureDataForV2 build a block sig data for V2
+// It reuses the hotstuff.BlockSignatureData type to create the sig data without filling the RandomBeaconSigners field and
+// the AggregatedRandomBeaconSig field, so that the packer can be reused by both V2 and V3 to pack QC's sig data.
```

Comment:
Adding this to explain why V2 and V3 can both use the `ConsensusSigDataPacker` to pack the sig data